### PR TITLE
Cleanup bugfiler.js

### PR DESCRIPTION
**Pull request by jgraham**
Changes split out from the autoclassification UI rewrite branch, and
are just minor cleanup of the existing code.

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/treeherder/2151)

### CODE REVIEW

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by jgraham** [comment]
@wlach You asked for this to be split out; I'm not convinced it's worthwhile (since it creates a new intermediate state to test). What do you think?

**Comment by wlach** [comment]
I like splitting out smaller changes like this as it makes it easier to review the larger patch. LGTM!

File: ui/js/controllers/bugfiler.js [hunk 1]
```diff
@@ -194,34 +196,39 @@ treeherder.controller('BugFilerCtrl', [
...
+                .then((response) => {
+                    var data = response.data;
+                    if (data.failure) {
                         var errorString = "";
-                        for (var i = 0; i < json.data.failure.length; i++) {
-                            errorString += json.data.failure[i];
+                        for (var i = 0; i < data.failure.length; i++) {
+                            errorString += data.failure[i];
                         }
                         errorString = JSON.parse(errorString);
```
Comment (wlach) [inline ui/js/controllers/bugfiler.js]: While you're here, you could simplify this to:

```js
var errorString = JSON.parse(data.failure.join(''));
```

Comment (wlach) [inline ui/js/controllers/bugfiler.js]: (maybe also rename from errorString to error)

File: ui/js/controllers/bugfiler.js [hunk 2]
```diff
@@ -235,7 +242,8 @@ treeherder.controller('BugFilerCtrl', [
                         window.open("https://bugzilla.mozilla.org/show_bug.cgi?id=" + json.data.success);
```
Comment (wlach) [inline ui/js/controllers/bugfiler.js]: I think you accidentally broke this codepath (see the eslint errors in travis)

**Comment by jgraham** [comment]
@wlach OK with the changes?

**Comment by wlach** [comment]
Looks good now, yeah. I assume you've done some basic testing? If so I'd say we're good to land.

**Comment by jgraham** [comment]
Hard to test without actually filing bugs, but it seems to load OK, and I guess it's easy to revert.

**Pull request merged**